### ISSUE TITLE
During checkout SR users are unable to tell which Shipping Address is default, additional or selected

### ISSUE THREAD

**Issue by developpeurweb** [SAP/spartacus#13724]
**Describe the bug**
- Screen reader users are unable to tell which address is default and which one is an additional address.
- Screen reader users are unable to tell which address is selected
- Action links are being described as links, because they are semantically links. They should be `<button>`s styled as links, they don't have an `href` to be followed.

**Tell us the version of Spartacus**
- Library version: 4.0.0

**To Reproduce**
1. Log in as a registered user
2. Add items to cart and start checkout process.
3. Open any Screen Reader (VO, NVDA, JAWS).
4. Tab to address cards and listen to screen reader feedback.
5. Hear absence of differentiation between addresses.

**Expected behavior**
- Screen reader users can hear which one is the default address as soon as they tab into it.
- Screen reader users can hear a differentiator for non-default addresses.
- Screen reader users can hear which address is selected, either the default or an additional address
- Use as per screenshot below `role="region" aria-label="Default shipping address" tabindex="0"` and 
`role="region" aria-label="Additional shipping address" tabindex="0"`
- Change all `<a tabindex="0" class="cx-card-link card-link btn-link cx-action-link">...</a>` to `<button class="cx-card-link cx-action-link"> ... </button>`
- As per screenshot below, use `aria-describedby` and an ID to associate the `<button>` with the "Selected" text state. E.g. `<div class="card-header" id="selectedTxt"> Selected </div>` ... ... `<button aria-describedby="selectedTxt" class="cx-card-link cx-action-link">Ship to this address</button>`

**Screenshots**
Expected screen reader vocalization:
1. Suggested Code
![Screen Shot 2021-09-15 at 9 11 24 AM](https://user-images.githubusercontent.com/15113219/133442176-1ea2430a-3fa4-444b-af54-6cf43c36dc61.png)
2. Keyboard navigation sequence (Tab key)
![Screen Shot 2021-09-15 at 9 17 54 AM](https://user-images.githubusercontent.com/15113219/133442238-88a4295b-00b6-4524-b74a-cfe0634213c8.png)
3.  Keyboard navigation sequence (Tab key)
![Screen Shot 2021-09-15 at 9 19 34 AM](https://user-images.githubusercontent.com/15113219/133442369-f9ad997b-62f0-4b4b-bcae-477777cffaf2.png)

4.  Keyboard navigation sequence (Tab key)
![Screen Shot 2021-09-15 at 9 16 35 AM](https://user-images.githubusercontent.com/15113219/133442480-31acd615-5ebe-4581-9e40-1946be1d2879.png)

5.  Keyboard navigation sequence (Tab key)
![Screen Shot 2021-09-15 at 9 18 41 AM](https://user-images.githubusercontent.com/15113219/133442528-d31ae7a3-5b43-4e43-a8b0-2649078f39b7.png)

6. Landmark navigation
![landmarks2021-09-15_09-32-55](https://user-images.githubusercontent.com/15113219/133443038-991f4a4c-83e3-4e40-b2c5-84c658e4fa2c.jpg)

**Comment by KateChuen** [comment]
Tested on develop branch of spartacus using installation script
More extensive testing (on other screenreaders and devices) will be done with RC.

Question: Should it be consistent with the payment checkout step? @developpeurweb 

1) Although the user can differentiate between the default and non-default ones on both payment and address, let me point out the difference:. (Here I like the payment one better)
Payment: It says "Additional payment method"
Address: It just says "Main"


![AddressAdditional](https://user-images.githubusercontent.com/45409515/149392916-07cd67ed-7196-4686-bb3a-5598a8c08ca6.png)


2) In both cases, the user can hear which one is selected. But it's still different. (Here I like the Address one better)
Payment: It says "Selected" when the focus is on the "Use this payment"
Address: It says "Selected" when the focus is on the address and also on the "Ship to this address"
![AddressDefaultSelected](https://user-images.githubusercontent.com/45409515/149393241-9fe55fce-e4f8-43fa-9a1e-b5dba7a06c5b.png)

![AddressSelected](https://user-images.githubusercontent.com/45409515/149393098-42f3dad4-6ef9-4064-b08d-0451b7dcec12.png)




3) I'm not fond of hearing the whole address again when the focus is on the "Ship to this address" button. 
![AddressDefaultSelected1](https://user-images.githubusercontent.com/45409515/149393246-89c12b03-5015-46c1-92aa-0f348b5cb085.png)
![AddressSelected1](https://user-images.githubusercontent.com/45409515/149393163-a6adbc8c-d6ed-43ae-9d7f-330908d4ff2a.png)

**Comment by BenjaminEmiliani** [comment]
Hi @KateChuen I think you might have tested this in an older version of the branch. I recently had to make a small fix which in now in develop. So you could you please retest this with latest version of develop branch?

**Comment by KateChuen** [comment]
Retested, it's consistent with Payment now. 
![ShipAddDefault](https://user-images.githubusercontent.com/45409515/149571854-5884c4ba-a4f5-416d-ab26-bd69581857c9.png)

![ShipAddButton](https://user-images.githubusercontent.com/45409515/149571848-c55b8cce-8080-43c9-84a8-a0924337098e.png)

![ShipAddSelected](https://user-images.githubusercontent.com/45409515/149571857-38b399ff-663d-4921-83dd-0b9fe781e932.png)
![AdditionalAdd1](https://user-images.githubusercontent.com/45409515/149571859-9364d051-6707-4871-8281-f0c74addaa48.png)
![ShipAddButton1](https://user-images.githubusercontent.com/45409515/149571862-63ed5d3c-05e2-4057-8ba5-68e9db9e04d5.png)
